code/game/objects/structures/displaycase.dm

```diff
@@ -342,3 +362,137 @@
 	name = initial(I.name)
 	icon = initial(I.icon)
 	icon_state = initial(I.icon_state)
+
+/obj/structure/displaycase/forsale
+	name = "vend-a-tray"
+	icon = 'icons/obj/stationobjs.dmi'
+	icon_state = "laserbox0"
+	desc = "A display case with an ID-card swiper. Use your ID to purchase the contents."
+	density = FALSE
+	max_integrity = 100
+	req_access = list(ACCESS_KITCHEN)
+	showpiece_type = /obj/item/reagent_containers/food
+	alert = FALSE //No, we're not calling the fire department because someone stole your cookie.
+	glass_fix = FALSE //Fixable with tools instead.
+	var/sale_price = 20
+	var/datum/bank_account/payments_acc = null
+
+/obj/structure/displaycase/forsale/update_icon()	//remind me to fix my shitcode later
+	var/icon/I
+	if(open)
+		I = icon('icons/obj/stationobjs.dmi',"laserboxb0")
+	else
+		I = icon('icons/obj/stationobjs.dmi',"laserbox0")
+	if(!showpiece && !open)
+		I = icon('icons/obj/stationobjs.dmi',"laserbox_open")
+	if(broken)
+		I = icon('icons/obj/stationobjs.dmi',"laserbox_broken")
+	if(showpiece)
+		var/icon/S = getFlatIcon(showpiece)
+		S.Scale(17,17)
+		I.Blend(S,ICON_UNDERLAY,8,12)
+	src.icon = I
+	return
+
+/obj/structure/displaycase/forsale/attackby(obj/item/I, mob/living/user, params)
+	if(isidcard(I))
+		//Card Registration
+		var/obj/item/card/id/potential_acc = I
+		if(!potential_acc.registered_account)
+			to_chat(user, "<span class='warning'>This ID card has no account registered!</span>")
+			return
+		if(payments_acc && payments_acc != potential_acc.registered_account)
+			to_chat(user, "<span class='warning'>This Vend-a-tray is already registered!</span>")
+			return
+		if(!payments_acc && potential_acc.registered_account)
+			payments_acc = potential_acc.registered_account
+			playsound(src, 'sound/machines/click.ogg', 20, TRUE)
+			to_chat(user, "<span class='notice'>Vend-a-tray registered. Use your ID on grab intent to change the sale price, or disarm intent to open the tray.</span>")
+			return
+		//Buying the contained item with the ID.
+		switch(user.a_intent)
+			if(INTENT_HELP)
+				if(!showpiece)
+					to_chat(user, "<span class='notice'>There's nothing for sale.</span>")
+					return TRUE
+				if(broken)
+					to_chat(user, "<span class='notice'>[src] appears to be broken.</span>")
+					return TRUE
+				var/confirm = alert(user, "Purchase [showpiece] for [sale_price]?", "Purchase?", "Confirm", "Cancel")
+				if(confirm == "Cancel")
+					return TRUE

```

Comment:
This alert is a bit of a troublemaker. You'll need to recheck user proximity and the ID card stuff as well before finalizing the purchase.